frontend/public/components/modals/cluster-update-modal.tsx

```diff
@@ -0,0 +1,99 @@
+/* eslint-disable no-undef, no-unused-vars */
+
+import * as _ from 'lodash-es';
+import * as React from 'react';
+
+import { createModalLauncher, ModalTitle, ModalBody, ModalSubmitFooter, ModalComponentProps } from '../factory/modal';
+import { Dropdown, PromiseComponent } from '../utils';
+import { k8sPatch, K8sResourceKind } from '../../module/k8s';
+
+class ClusterUpdateModal extends PromiseComponent {
+  public state: ClusterUpdateModalState;
+
+  constructor(public props: ClusterUpdateModalProps) {
+    super(props);
+    this._submit = this._submit.bind(this);
+    this._cancel = props.cancel.bind(this);
+    this._change = this._change.bind(this);
+    this.state.selectedVersion = 0;
+  }
+
+  _submit(e) {
+    e.preventDefault();
+    const { cv } = this.props;
+    const {selectedVersion} = this.state;
+    const desiredUpdate = this.props.availableUpdates[selectedVersion];
+    const patch = {
+      spec: { desiredUpdate },
+    };
+    this.handlePromise(k8sPatch('ClusterVersion', cv, patch)).then(this.props.close);
+  }
+
+  _cancel() {
+    this.props.close();
+  }
+
+  _change(selectedVersion) {
+    this.setState({selectedVersion});
+  }
+
+  render() {
+    const {currentVersion, availableUpdates} = this.props;
+    const {selectedVersion} = this.state;
+    const dropdownItems = _.map(availableUpdates, 'version');
+    const dropdownTitle = _.get(availableUpdates[selectedVersion], 'version');
+    return <form onSubmit={this._submit} name="form">
+      <ModalTitle>Update Cluster</ModalTitle>
+      <ModalBody>
+        <p>
+          For more detailed documentation on specific versions, &nbsp;
+          <a
+            className="co-external-link"
+            href="https://github.com/openshift/origin/releases"
+            rel="noopener noreferrer"
+            target="_blank"
+          >
```

Comment:
```jsx
<ExternalLink href="https://github.com/openshift/origin/releases" text="view release notes." />
```

Reply:
I'm not sure if we will know where to point this link to for different versions.... We'll need to figure this out.

Reply:
Yeah, I just put something there so that it's not a dead-end.

Reply:
I'm debating if we should take this out until we have a good link. I'm worried we'll forget to update it later.

Reply:
I still think we should have some sort of introductory text there. But I'm no good at copy, haha.